### PR TITLE
Update Core widgets provider display name

### PR DESCRIPTION
**Pull request by EricJohnson327**
## Summary of the pull request
When adding a new widget, the "Core" provider will now display "Core (Preview)"

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: src/Package.appxmanifest [hunk 1]
```diff
@@ -66,7 +61,7 @@
           </uap3:AppExtension>
         </uap3:Extension>
         <uap3:Extension Category="windows.appExtension">
-          <uap3:AppExtension Name="com.microsoft.windows.widgets" DisplayName="Core" Id="CoreWidgetProvider" PublicFolder="Public">
+          <uap3:AppExtension Name="com.microsoft.windows.widgets" DisplayName="ms-resource:WidgetsDisplayNameDev" Id="CoreWidgetProvider" PublicFolder="Public">
```
Comment (krschau) [inline src/Package.appxmanifest:64]:
```suggestion
          <uap3:AppExtension Name="com.microsoft.windows.widgets" DisplayName="ms-resource:WidgetProviderDisplayNameDev" Id="CoreWidgetProvider" PublicFolder="Public">
```

I think calling it WidgetProvider will be more clear.

File: src/Strings/en-us/Resources.resw [hunk 2]
```diff
@@ -210,4 +210,16 @@
...
+    <comment>The display name for core widgets provider</comment>
+  </data>
+  <data name="WidgetsDisplayNameCanary" xml:space="preserve">
+    <value>Core (Canary)</value>
+    <comment>The display name for core widgets provider</comment>
+  </data>
+  <data name="WidgetsDisplayNameStable" xml:space="preserve">
+    <value>Core (Preview)</value>
+    <comment>The display name for core widgets provider</comment>
+  </data>
```
Comment (krschau) [inline src/Strings/en-us/Resources.resw:224]:
```suggestion
  <data name="WidgetProviderDisplayNameDev" xml:space="preserve">
    <value>Core (Dev)</value>
    <comment>The display name for core widgets provider</comment>
  </data>
  <data name="WidgetProviderDisplayNameCanary" xml:space="preserve">
    <value>Core (Canary)</value>
    <comment>The display name for core widgets provider</comment>
  </data>
  <data name="WidgetProviderDisplayNameStable" xml:space="preserve">
    <value>Core (Preview)</value>
    <comment>The display name for core widgets provider</comment>
  </data>
```

File: Build.ps1 [hunk 3]
```diff
@@ -162,8 +170,13 @@ Try {
...
       if ($extension.Attribute("Category").Value -eq "windows.appExtension") {
         $appExtension = $extension.Element($uapAppExtension)
-        if ($appExtension.Attribute("Name").Value -eq "com.microsoft.devhome") {
-          $appExtension.Attribute("DisplayName").Value = "ms-resource:AppDisplayNameDev"
+        switch ($appExtension.Attribute("Name").Value) {
+          "com.microsoft.devhome" {
+            $appExtension.Attribute("DisplayName").Value = "ms-resource:AppDisplayNameDev"
+          }
+          "com.microsoft.windows.widgets" {
+            $appExtension.Attribute("DisplayName").Value = "ms-resource:WidgetsDisplayNameDev"
```
Comment (krschau) [inline Build.ps1:178]:
```suggestion
            $appExtension.Attribute("DisplayName").Value = "ms-resource:WidgetProviderDisplayNameDev"
```

File: Build.ps1 [hunk 4]
```diff
@@ -81,17 +81,20 @@ Try {
...
       $newPackageName = "Microsoft.Windows.DevHome"
       $newPackageDisplayName = "Dev Home (Preview)"
       $newAppDisplayNameResource = "ms-resource:AppDisplayNameStable"
+      $newWidgetsDisplayName = "ms-resource:WidgetsDisplayNameStable"
     } elseif ($AzureBuildingBranch -ieq "staging") {
       $buildRing = "Canary"
       $newPackageName = "Microsoft.Windows.DevHome.Canary"
       $newPackageDisplayName = "Dev Home (Canary)"
       $newAppDisplayNameResource = "ms-resource:AppDisplayNameCanary"
+      $newWidgetsDisplayName = "ms-resource:WidgetsDisplayNameCanary"
```
Comment (krschau) [inline Build.ps1:97]:
```suggestion
      $newWidgetProviderDisplayName = "ms-resource:WidgetProviderDisplayNameCanary"
```

File: Build.ps1 [hunk 5]
```diff
@@ -81,17 +81,20 @@ Try {
...
     $newPackageDisplayName = $null
     $newAppDisplayNameResource = $null
+    $newWidgetsDisplayName = $null
 
     if ($AzureBuildingBranch -ieq "release") {
       $buildRing = "Stable"
       $newPackageName = "Microsoft.Windows.DevHome"
       $newPackageDisplayName = "Dev Home (Preview)"
       $newAppDisplayNameResource = "ms-resource:AppDisplayNameStable"
+      $newWidgetsDisplayName = "ms-resource:WidgetsDisplayNameStable"
```
Comment (krschau) [inline Build.ps1:91]:
```suggestion
      $newWidgetProviderDisplayName = "ms-resource:WidgetProviderDisplayNameStable"
```

File: Build.ps1 [hunk 6]
```diff
@@ -81,17 +81,20 @@ Try {
     $newPackageName = $null
     $newPackageDisplayName = $null
     $newAppDisplayNameResource = $null
+    $newWidgetsDisplayName = $null
```
Comment (krschau) [inline Build.ps1:84]:
```suggestion
    $newWidgetProviderDisplayName = $null
```

File: Build.ps1 [hunk 7]
```diff
@@ -121,8 +124,13 @@ Try {
...
         if ($extension.Attribute("Category").Value -eq "windows.appExtension") {
           $appExtension = $extension.Element($uapAppExtension)
-          if ($appExtension.Attribute("Name").Value -eq "com.microsoft.devhome") {
-            $appExtension.Attribute("DisplayName").Value = $newAppDisplayNameResource
+          switch ($appExtension.Attribute("Name").Value) {
+            "com.microsoft.devhome" {
+              $appExtension.Attribute("DisplayName").Value = $newAppDisplayNameResource
+            }
+            "com.microsoft.windows.widgets" {
+              $appExtension.Attribute("DisplayName").Value = $newWidgetsDisplayName
```
Comment (krschau) [inline Build.ps1:132]:
```suggestion
              $appExtension.Attribute("DisplayName").Value = $newWidgetProviderDisplayName
```

**Review by krschau — CHANGES_REQUESTED** [review]
I quickly went from thinking the code suggestions were helpful to obnoxious. Sorry.

**Comment by EricJohnson327** [comment]
> I quickly went from thinking the code suggestions were helpful to obnoxious. Sorry.

Haha!  No worries.  I just learned that you can batch suggested changes to do one commit.  Neat :)

**Review by krschau — APPROVED** [review]

**Review by guimafelipe — APPROVED** [review]

**Comment by luislhg** [comment]
I think there might be an error in the resources namings in this PR
Please check printscreen here https://github.com/microsoft/devhome/issues/1187#issuecomment-1601865632

**Pull request merged**